src/riscv/lib/src/pvm/linux/signals.rs

```diff
@@ -114,25 +197,12 @@ impl TryFrom<u64> for TkillSignal {
 }
 
 /// An address of a signal action in the VM memory
-#[derive(Clone, Copy)]
-pub struct SignalActionPtr(pub Option<NonZeroU64>);
-
-impl SignalActionPtr {
-    /// Extract the address of the signal action in the VM memory
-    pub fn address(&self) -> Option<u64> {
-        self.0.map(|nz| nz.get())
-    }
```

Comment:
You can keep this method to avoid accessing the unnamed struct field below.

```rust
    pub fn address(&self) -> Option<u64> {
        self.0.map(|addr| addr.to_machine_address())
    }
```

Instead of `signal.0`, you can then `signal.address()` in the edits below. The latter conveys the function more clearly.